doc/source/conf.py

```diff
@@ -1,5 +1,9 @@
+import os
+import sys
 from typing import List
 
+sys.path.insert(0, os.path.abspath(os.path.join("..", "..", "src")))
```

Comment:
Is there a way to do this without having to change `sys.path`? Also, I think it would be better if you used a path relative to this file instead of the current working directory of the the shell, so something like:

```python
srcdir = pathlib.Path(__file__).resolve().parent.parent / "src"
# do whatever needs to be done with srcdir
```

Reply:
@WillAyd there's two standard ways to work with autodoc as per [the official docs](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#ensuring-the-code-can-be-imported).

1. Do an install of the package in your env and then run sphinx.
2. Add the sources to the path and then run sphinx.

The latter just seemed less frictional to me as an outside contributor and hence I went ahead with it. I'm not sure if there's a CI workflow to publish the docs or if they're automatically generated. If it's the former and I can assume that the package will be installed in the environment before the docs are generated, I can remove the `sys.path.insert`. Both workflows are perfectly standard however.

Reply:
> I'm not sure if there's a CI workflow to publish the docs or if they're automatically generated

This is all handled by readthedocs.com, which looks for changes to this repo and republishes the docs when pushes are made to main. If you sign up for an account there I'd be happy to add you to the project; its probably helpful for what you are trying to do

Reply:
Sounds good. "Priyansh121096" is my readthedocs username.